transport/http2_client.go

```diff
@@ -1082,6 +1105,26 @@ func (t *http2Client) controller() {
 			case <-t.shutdownChan:
 				return
 			}
+		case <-timer.C:
+			t.mu.Lock()
+			ns := len(t.activeStreams)
+			t.mu.Unlock()
+			// Get the activity counter value and reset it.
+			a := atomic.SwapUint64(&t.activity, 0)
+			if a > 0 || (!t.kp.PermitWithoutStream && ns < 1) {
```

Comment:
This doesn't seem to follow this part of the spec:
> Since keepalive is not occurring on HTTP/2 connections without any streams, there will be a higher chance of failure for new RPCs following a long period of inactivity. To reduce the tail latency for these RPCs, it is important to not reset the `keepalive time` when a connection becomes active; if a new stream is created and there has been greater than `keepalive time` since the last read byte, then a keepalive PING should be sent (ideally before the HEADERS frame). Doing so detects the broken connection with a latency of `keepalive timeout` instead of `keepalive time + timeout`.